stores/wallet.ts

```diff
@@ -88,14 +177,335 @@ export const useWalletStore = defineStore('wallet', () => {
+
+          default:
+        }
+      }
+      throw error
+    }
+  }
+
+  function getEmailAlreadyUsedErrorData ({
+    email,
+    walletAddress,
+    boundEVMWallet,
+    boundLikeWallet,
+    loginMethod
+  }: {
+    email: string
+    walletAddress: string
+    boundEVMWallet?: string
+    boundLikeWallet?: string
+    loginMethod: string
+  }) {
+    const shouldMigrate = !boundEVMWallet && !!boundLikeWallet
+    return {
+      statusCode: 401,
+      data: {
+        level: 'warning',
+        title: shouldMigrate
+          ? $t('account_register_error_email_already_used_migrate_title')
+          : $t('account_register_error_email_already_used_by_wallet_title'),
+        description: getEmailAlreadyUsedErrorMessage({
+          email,
+          evmWallet: boundEVMWallet,
+          likeWallet: boundLikeWallet
+        }),
+        tags: [
+          { label: loginMethod, icon: 'i-material-symbols-login-rounded', class: 'font-mono' },
+          { label: walletAddress, icon: 'i-material-symbols-key-outline-rounded', class: 'font-mono' }
+        ],
+        actions: shouldMigrate
+          ? [{
+              label: $t('account_register_error_contact_support'),
+              color: 'secondary',
+              variant: 'subtle',
+              onClick: async () => {
+                await navigateTo(getLikeCoinV3BookMigrationSiteURL.value({ utmSource: 'login_email_already_used' }), {
+                  external: true,
+                  open: { target: '_blank' }
+                })
+              }
+            }]
+          : []
+      }
+    }
+  }
+
+  function getEmailAlreadyUsedErrorMessage ({
+    email,
+    evmWallet,
+    likeWallet
+  }: {
+    email: string
+    evmWallet?: string
+    likeWallet?: string
+  }) {
+    if (evmWallet) {
+      return $t('account_register_error_email_already_used_with_evm_wallet', { email, evmWallet })
+    }
+    if (likeWallet) {
+      return $t('account_register_error_email_already_used_with_like_wallet', { email, likeWallet })
+    }
+    return $t('account_register_error_email_already_used', { email })
+  }
+
+  async function register ({
+    walletAddress,
+    email: prefilledEmail,
+    loginMethod,
+    magicUserId,
+    magicDIDToken
+  }: {
+    walletAddress: string
+    email?: string
+    loginMethod: string
+    magicUserId?: string
+    magicDIDToken?: string
+  }) {
+    let tempAccountId = generateAccountIdFromWalletAddress(walletAddress)
+    try {
+      await fetchUserRegisterCheck({ accountId: tempAccountId })
+    } catch (error) {
+      if (error instanceof FetchError && error.data?.error === 'USER_ALREADY_EXIST') {
+        tempAccountId = error.data.alternative as string
+      } else {
+        throw error
+      }
+    }
+
+    const startTime = Date.now()
+    let hasError = false
+    const payload = {
+      accountId: tempAccountId,
+      email: prefilledEmail
+    }
+
+    // Loop until registration is successful, user cancels or timeout
+    do {
```

Comment:
The loop is too large, we should only loop the part where we keep poping up `RegistrationModal` on cancel or error. We should keep looping part minimal and with clear intent, and no other API error should be included into the looping condition